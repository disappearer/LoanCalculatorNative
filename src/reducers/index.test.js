import reducer from './index';
import {
  FETCH_CONSTRAINTS_REQUEST,
  FETCH_CONSTRAINTS_SUCCESS,
  FETCH_CONSTRAINTS_FAIL,
  FETCH_OFFER_REQUEST,
  FETCH_OFFER_SUCCESS,
  FETCH_OFFER_FAIL,
  SELECT_AMOUNT,
  SELECT_TERM
} from '../actions';

const defaultState = {
  amountInterval: null,
  termInterval: null,
  amount: 0,
  term: 0,
  totalCostOfCredit: 0,
  totalRepayableAmount: 0,
  monthlyPayment: 0,
  isFetchingConstraints: true,
  isFetchingOffer: true,
  error: null
};

const constraints = {
  amountInterval: { min: 10, max: 2000, step: 10, defaultValue: 400 },
  termInterval: { min: 3, max: 30, step: 1, defaultValue: 15 }
};

const offer = {
  totalPrincipal: '400',
  term: '15',
  totalCostOfCredit: 40,
  totalRepayableAmount: 480,
  monthlyPayment: 32
};

describe('Reducer', () => {
  it('Returns default state', () => {
    expect(reducer(undefined)).toEqual(defaultState);
  });

  it('Handles action FETCH_CONSTRAINTS_REQUEST', () => {
    const action = { type: FETCH_CONSTRAINTS_REQUEST };
    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  it('Handles action FETCH_CONSTRAINTS_SUCCESS', () => {
    const action = { type: FETCH_CONSTRAINTS_SUCCESS, constraints };
    expect(
      reducer({ ...defaultState, isFetchingConstraints: true }, action)
    ).toMatchSnapshot();
  });

  it('Handles action FETCH_CONSTRAINTS_FAIL', () => {
    const error = new Error('Fetching constraints failed');
    const action = { type: FETCH_CONSTRAINTS_FAIL, error };
    expect(
      reducer({ ...defaultState, isFetchingConstraints: true }, action)
    ).toMatchSnapshot();
  });

  it('Handles action FETCH_OFFER_REQUEST', () => {
    const action = { type: FETCH_OFFER_REQUEST };
    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  it('Handles action FETCH_OFFER_SUCCESS', () => {
    const action = { type: FETCH_OFFER_SUCCESS, offer };
    expect(
      reducer({ ...defaultState, isFetchingOffer: true }, action)
    ).toMatchSnapshot();
  });

  it('Handles action FETCH_OFFER_FAIL', () => {
    const error = new Error('Fetching offer failed');
    const action = { type: FETCH_OFFER_FAIL, error };
    expect(
      reducer({ ...defaultState, isFetchingOffer: true }, action)
    ).toMatchSnapshot();
  });

  it('Handles action SELECT_AMOUNT', () => {
    const amount = 301;
    const action = { type: SELECT_AMOUNT, amount };
    expect(reducer(defaultState, action)).toMatchSnapshot();
  });

  it('Handles action SELECT_TERM', () => {
    const term = 30;
    const action = { type: SELECT_TERM, term };
    expect(reducer(defaultState, action)).toMatchSnapshot();
  });
});
