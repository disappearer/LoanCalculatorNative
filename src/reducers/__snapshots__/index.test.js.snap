// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer Handles action FETCH_CONSTRAINTS_FAIL 1`] = `
Object {
  "amount": 0,
  "amountInterval": null,
  "error": [Error: Fetching constraints failed],
  "isFetchingConstraints": false,
  "isFetchingOffer": true,
  "monthlyPayment": 0,
  "term": 0,
  "termInterval": null,
  "totalCostOfCredit": 0,
  "totalRepayableAmount": 0,
}
`;

exports[`Reducer Handles action FETCH_CONSTRAINTS_REQUEST 1`] = `
Object {
  "amount": 0,
  "amountInterval": null,
  "error": null,
  "isFetchingConstraints": true,
  "isFetchingOffer": true,
  "monthlyPayment": 0,
  "term": 0,
  "termInterval": null,
  "totalCostOfCredit": 0,
  "totalRepayableAmount": 0,
}
`;

exports[`Reducer Handles action FETCH_CONSTRAINTS_SUCCESS 1`] = `
Object {
  "amount": 400,
  "amountInterval": Object {
    "defaultValue": 400,
    "max": 2000,
    "min": 10,
    "step": 10,
  },
  "error": null,
  "isFetchingConstraints": false,
  "isFetchingOffer": true,
  "monthlyPayment": 0,
  "term": 15,
  "termInterval": Object {
    "defaultValue": 15,
    "max": 30,
    "min": 3,
    "step": 1,
  },
  "totalCostOfCredit": 0,
  "totalRepayableAmount": 0,
}
`;

exports[`Reducer Handles action FETCH_OFFER_FAIL 1`] = `
Object {
  "amount": 0,
  "amountInterval": null,
  "error": [Error: Fetching offer failed],
  "isFetchingConstraints": true,
  "isFetchingOffer": false,
  "monthlyPayment": 0,
  "term": 0,
  "termInterval": null,
  "totalCostOfCredit": 0,
  "totalRepayableAmount": 0,
}
`;

exports[`Reducer Handles action FETCH_OFFER_REQUEST 1`] = `
Object {
  "amount": 0,
  "amountInterval": null,
  "error": null,
  "isFetchingConstraints": true,
  "isFetchingOffer": true,
  "monthlyPayment": 0,
  "term": 0,
  "termInterval": null,
  "totalCostOfCredit": 0,
  "totalRepayableAmount": 0,
}
`;

exports[`Reducer Handles action FETCH_OFFER_SUCCESS 1`] = `
Object {
  "amount": 0,
  "amountInterval": null,
  "error": null,
  "isFetchingConstraints": true,
  "isFetchingOffer": false,
  "monthlyPayment": 32,
  "term": 0,
  "termInterval": null,
  "totalCostOfCredit": 40,
  "totalRepayableAmount": 480,
}
`;

exports[`Reducer Handles action SELECT_AMOUNT 1`] = `
Object {
  "amount": 301,
  "amountInterval": null,
  "error": null,
  "isFetchingConstraints": true,
  "isFetchingOffer": true,
  "monthlyPayment": 0,
  "term": 0,
  "termInterval": null,
  "totalCostOfCredit": 0,
  "totalRepayableAmount": 0,
}
`;

exports[`Reducer Handles action SELECT_TERM 1`] = `
Object {
  "amount": 0,
  "amountInterval": null,
  "error": null,
  "isFetchingConstraints": true,
  "isFetchingOffer": true,
  "monthlyPayment": 0,
  "term": 30,
  "termInterval": null,
  "totalCostOfCredit": 0,
  "totalRepayableAmount": 0,
}
`;
